kickstarter/android-oss
# Migrate Horizontal rewards into the Project Page

# What ❓
- Migrated the great work @eoji did with the `Checkout POC` into the KSR codebase for the first step towards `Horizontal Rewards`. On `internal` builds user's have the ability to toggle on/off the horizontal rewards from the `InternalToolsActivity`.
- Great collaboration with @Scollaco 🎉🎉
- We created the `RewardFragment`, `RewardFragmentViewModel`, `HorizontalRewardViewHolder`, `HorizontalRewardAdapter`,`RewardFragmentViewModelTest`. Also we updated the tests for `ProjectViewModelTest`


# How to QA? 🤔
- When you launch the app with an internal build `open the navigation drawer -> InternalTools -> FeatureFlagActivity -> Toggle on or off`.
- When enabled navigate to the project page and you will no longer see the list of rewards below the project. When disabled you will see the project page in it's original state.

# Story 📖

[Migrate Horizontal Rewards](https://trello.com/c/GTubQ0dO/1305-migrate-checkout-work-from-prototype-to-the-real-app)

# See 👀
| Feature flag disabled | Feature flag enabled|
| --- | --- |
| <img src="https://user-images.githubusercontent.com/16387538/54951153-50ce8800-4f19-11e9-933e-2e18ea5609e8.gif"> | <img src="https://user-images.githubusercontent.com/16387538/54951152-50ce8800-4f19-11e9-9760-b5a50ca9bbaa.gif"> |
